File: Python/no. checking/disarium.py
#Disarium No.
n=int(input("Enter the no. = "))    #153
x=n
c=len(str(n))                       #3
s=0
while(n!=0):
    s+=(n%10)**c
    c-=1
    n//=10
if s==x:
    print("Disarium No.")
else:
    print("not Disarium No.")

File: C Programms/No. Cheking In Range/disarium.c
//WAP to check Disarium No.
#include<stdio.h>
void main()
{
    int n,p,i,s,c,x,r;
    printf("Enter the range = ");
    scanf("%d",&r);
    printf("Disarium no. from 1 to %d are ...\n",r);
    for(n=1;n<=r;n++)
    {
        for(x=n,c=0;x>0;x/=10)
            c++;
        for(x=n,s=0;x!=0;x/=10,c--)
        {
            for(i=1,p=1;i<=c;i++)
                p*=x%10;
            s+=p;
        }
        if(s==n)
            printf("%d ",n);
    }
}